### ISSUE TITLE
Shader loading fails when using a jar version of the library

### ISSUE THREAD

**Issue by laserallan** [kotlin-graphics/imgui#2]
I get an exception in Program.kt around line 294:
        val lines = File(url.toURI()).readLines()

Looking at the url it looks like: 

`jar:file:/C:/Users/<myusername>/.m2/repository/com/github/kotlin-graphics/imgui/-v1.51-gfcc47f9-11/imgui--v1.51-gfcc47f9-11.jar!/shader.vert`

It might be an issue exposed when grabbing imgui as a jar file from maven rather than referring directly to the code. It seems Files can't be created from a path pointing into a jar. 
Using url.openStream() seem to solve this in my local tests (I don't have an environment allowing me to rebuild the imgui sources at this point so I have not tried it on the imgui code)

**Comment by laserallan** [comment]
Disregard. This error happened in uno-sdk

**Comment by elect86** [comment]
Hi,

I'm interested in pursuing the issue also if this regards uno-sdk..

I'm used to work using dependencies managed by Gradle.. so I don't have the possibility normally to test other setups and this would be a perfect chance to do that

Let me know if you still have problems, we cant work on that :)

**Comment by laserallan** [comment]
It's certainly still a problem. I made a cowboy fix looking like this after
getting the source code for uno:

val url = ClassLoader.getSystemResource(path)
var str = url.openStream().bufferedReader()
//val lines = File(url.toURI()).readLines()
val lines = str.readLines()

Let me know if it is something you want me to clean up and make a pull
request out of (it needs to be changed in both functions called
createShader).

I have not gone deeper down this rabbit hole since I ran into some font
loading issue with imgui that blocked me from showing anything meaningful
so there might be other places where the same issue happens. Also, I have
not tried it on a source code build so I don't know if this works when
referring the source directly rather than the jar.

David
